crates/chain/tests/test_indexed_tx_graph.rs

```diff
@@ -1,19 +1,597 @@
+    };
+
+    // tx3 spends tx2 and gives a change back in trusted keychain. Confirmed at Block 2.
+    let tx3: TxTemplate<'_, _, ConfirmationHeightAnchor> = TxTemplate {
+        tx_name: "tx3",
+        spends: &[TxInTemplate::PrevTx("tx2", 0)],
+        outputs: &[TxOutTemplate {
+            value: 10000,
+            owned_spk: Some((TestKeychain::External, 1)),
+        }],
+        anchors: &[ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: 2,
+                hash: *local_chain.blocks().get(&2).unwrap(),
+            },
+            confirmation_height: 2,
+        }],
+        last_seen: None,
+    };
+
+    // tx4 is an external transaction receiving at untrusted keychain, unconfirmed.
+    let tx4: TxTemplate<'_, _, ConfirmationHeightAnchor> = TxTemplate {
+        tx_name: "tx4",
+        outputs: &[TxOutTemplate {
+            value: 20000,
+            owned_spk: Some((TestKeychain::Internal, 1)),
+        }],
+        last_seen: Some(100),
+        ..Default::default()
+    };
+
+    // tx5 is spending tx3 and receiving change at trusted keychain, unconfirmed.
+    let tx5: TxTemplate<'_, _, ConfirmationHeightAnchor> = TxTemplate {
+        tx_name: "tx5",
+        outputs: &[TxOutTemplate {
+            value: 15000,
+            owned_spk: Some((TestKeychain::External, 2)),
+        }],
+        last_seen: Some(100),
+        ..Default::default()
+    };
+
+    // tx6 is an irrelevant transaction confirmed at Block 3.
+    let tx6 = TxTemplate {
+        tx_name: "tx6",
+        anchors: &[ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: 3,
+                hash: *local_chain.blocks().get(&3).unwrap(),
+            },
+            confirmation_height: 3,
+        }],
+        last_seen: None,
+        ..Default::default()
+    };
+
+    let scenarios = [
+        Scenario {
+            name: "first test",
+            tx_templates: &[tx1, tx2, tx3, tx6],
+            exp_chain_txs: HashSet::default(),
+            exp_unspents: HashSet::default(),
+            exp_balance: Balance {
+                immature: 0,
+                trusted_pending: 0,
+                untrusted_pending: 0,
+                confirmed: 0,
+            },
+        },
+        Scenario {
+            name: "second test",
+            tx_templates: &[tx4, tx5],
+            exp_chain_txs: HashSet::default(),
+            exp_unspents: HashSet::default(),
+            exp_balance: Balance {
+                immature: 0,
+                trusted_pending: 0,
+                untrusted_pending: 0,
+                confirmed: 0,
+            },
+        },
+    ];
+
+    for scenario in scenarios {
+        let (temp_graph, temp_tx_ids) = init_graph(
+            graph.clone(),
+            graph.index.keychains().clone().into_iter().collect(),
+            scenario.tx_templates.iter(),
+        );
+        graph = temp_graph;
+        tx_ids.extend(temp_tx_ids);
+    }
```

Comment:
Looks like we aren't using `Scenario` for the purpose that they are designed for.

If there is no way to use it for these tests here, we can just not use it.

Reply:
This was my next step, and is what I am currently working on.